### PR TITLE
Add lvgl_8.1.0 support for raspberry_pi pico 

### PR DESCRIPTION
**Pull request by Rbb666**
## 拉取/合并请求描述：(PR description)

[
这段方括号里的内容是您**必须填写并替换掉**的，否则PR不可能被合并。**方括号外面的内容不需要修改，但请仔细阅读。**
The content in this square bracket must be filled in and replaced, otherwise, PR can not be merged. The contents outside square brackets need not be changed, but please read them carefully.

请在这里填写您的PR描述，可以包括以下之一的内容：为什么提交这份PR；解决的问题是什么，你的解决方案是什么；
Please fill in your PR description here, which can include one of the following items: why to submit this PR; what is the problem solved and what is your solution;

并确认并列出已经在什么情况或板卡上进行了测试。
And confirm in which case or board has been tested.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: bsp/raspberry-pico/applications/lvgl/SConscript [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+src = Glob('*.c')
+CPPPATH = [cwd]
+
+list = os.listdir(cwd)
+for d in list:
+    path = os.path.join(cwd, d)
+    if os.path.isfile(os.path.join(path, 'SConscript')):
+        group = group + SConscript(os.path.join(d, 'SConscript'))
+
+group += DefineGroup('LVGL-port', src, depend = ['BSP_USING_LVGL'], CPPPATH = CPPPATH)
```
Comment (mysterywolf) [inline bsp/raspberry-pico/applications/lvgl/SConscript:15]: 不要写成+= 要写成group = group + xxxx   python中 group+= 和 group= group+行为是不一样的

File: bsp/raspberry-pico/applications/lvgl/lv_port_disp.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+void lv_port_disp_init(void)
+{
+    rt_err_t result;
+
+    SPI_Init();
+    LCD_Init(HORIZONTAL);
+
+    /*Initialize `disp_buf` with the buffer(s).*/
+    lv_disp_draw_buf_init(&disp_buf, lv_disp_buf1, 0x00, DISP_BUF_SIZE);
```
Comment (mysterywolf) [inline bsp/raspberry-pico/applications/lvgl/lv_port_disp.c:39]: 0x00改成RT_NULL

File: bsp/raspberry-pico/applications/SConscript [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 
 group = DefineGroup('Applications', src, depend = [''], CPPPATH = CPPPATH)
 
-Return('group')
+list = os.listdir(cwd)
+for item in list:
+    if os.path.isfile(os.path.join(cwd, item, 'SConscript')):
+        group = group + SConscript(os.path.join(item, 'SConscript'))
+
+Return('group')
```
Comment (mysterywolf) [inline bsp/raspberry-pico/applications/SConscript:17]: 需要多加一个回车

File: bsp/raspberry-pico/applications/lvgl/SConscript (hunk 1 above)
Comment (mysterywolf) [reply bsp/raspberry-pico/applications/lvgl/SConscript:15]: 应该这么写，遍历放在最下面：
```python

group = DefineGroup('LVGL-port', src, depend = ['BSP_USING_LVGL'], CPPPATH = CPPPATH)

list = os.listdir(cwd)
for d in list:
    path = os.path.join(cwd, d)
    if os.path.isfile(os.path.join(path, 'SConscript')):
        group = group + SConscript(os.path.join(d, 'SConscript'))

```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

File: bsp/raspberry-pico/applications/lvgl/SConscript [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from building import *
+import os
+
+cwd = GetCurrentDir()
+group = []
+src = Glob('*.c')
+CPPPATH = [cwd]
+
+group = DefineGroup('LVGL-port', src, depend = ['BSP_USING_LVGL'], CPPPATH = CPPPATH)
```
[the remark this answers is not in the document]
Comment (mysterywolf) [reply bsp/raspberry-pico/applications/lvgl/SConscript:9]: 对的 port

**Review by Guozhanxin — APPROVED** [review]

**Comment by lliunix** [comment]
翻阅了提交的文件，发现并没有关于 LCD 型号和控制器的描述（只知道是240x240的spi屏幕，或许是st7789？），这很难应用呀。

**Pull request merged**